### ISSUE TITLE
[BUG] Incorrect subimages for TX files

### ISSUE THREAD

**Issue by zavinator** [OpenImageIO/oiio#2719]
**Describe the bug**
Traversing subimages in TX files (multipage TIF) cause an infinitive loop. 
It is not returning a correct ImageSpec for the given page.

**To Reproduce**
```
auto in = OIIO::ImageInput::open("normal.tif");
auto spec = in->spec();
int subimage = 0;
while (spec.format != OIIO::TypeUnknown) 
{
  //...
  spec = in->spec(++subimage); // next subimage
}
```

**Expected behavior**
It works correctly for multipart EXRs - the loop ends by returning OIIO::TypeUnknown format if subimage not exists.
It also looks that the spec is not updated after calling `in->spec(++subimage)`

**Evidence**
[normal.zip](https://github.com/OpenImageIO/oiio/files/5268474/normal.zip)

**Platform information:**
 - OIIO branch/version: 2.1.17
 - OS: Windows 10 x64
 - C++ compiler: VS2019 x64
 - Any non-default build flags when you build OIIO:

**Comment by lgritz** [comment]
Yes, I can reproduce.  Looking into it...

**Comment by lgritz** [comment]
A combination of a bug and a misunderstanding.

When a TIFF file is found to contain the tag `TIFFTAG_PIXAR_TEXTUREFORMAT`, it's recognized as a MIP-mapped texture, with the multiple TIFF directories interpreted as MIPmap levels of a single "subimage" (whereas without that tag, the image is treated as if it has multiple subimages, but they aren't MIPmapped).

The full signature of the spec() function is:  `ImageInput::spec(int subimage, int miplevel=0)`

I can't tell if you (a) want to walk through what OIIO considers subimages, and expect to see just ONE (which will have many MIP levels, but you aren't iterating over those in your example); or (b) if you want to walk through the MIP levels.

If (b), then you want your loop to look like this:

    int miplevel = 0;
    while (spec.format != OIIO::TypeUnknown) 
    {
      //...
      spec = in->spec(0, ++miplevel); // next MIP level
    }

However, if you really did want (a), then you are doing it correctly and there is a bug -- totally my fault -- in ImageInput::spec(), in the specific case where it's a "mip-mapped" TIFF file but you are passing an invalid subimage request. I will post a patch immediately, stay tuned...

**Comment by lgritz** [comment]
Proposed fix: #2723

**Comment by zavinator** [comment]
(a) is right :) 
I'm using the subimage for multipart exrs and I don't need to walk through the MIP levels...
